reflex/app.py

```diff
@@ -505,7 +505,10 @@ def add_page(
         # Check if the route given is valid
         verify_route_validity(route)
 
-        if route in self.unevaluated_pages and os.getenv(constants.RELOAD_CONFIG):
+        # TODO: this was broken?
```

Comment:
@reflex-dev is this intended behavior? I wanted to use `environment.RELOAD_CONFIG.get` here, but it breaks the tests. Take a look here: https://github.com/reflex-dev/reflex/pull/4255

Reply:
This should work as well
```diff
diff --git a/reflex/app.py b/reflex/app.py
index c4bbf8bf..40942d10 100644
--- a/reflex/app.py
+++ b/reflex/app.py
@@ -12,7 +12,6 @@ import inspect
 import io
 import json
 import multiprocessing
-import os
 import platform
 import sys
 import traceback
@@ -505,9 +504,9 @@ class App(MiddlewareMixin, LifespanMixin, Base):
         # Check if the route given is valid
         verify_route_validity(route)
 
-        # TODO: this was broken?
-        if route in self.unevaluated_pages and os.getenv(
-            environment.RELOAD_CONFIG.name
+        if (
+            route in self.unevaluated_pages
+            and environment.RELOAD_CONFIG.get is not None
         ):
             # when the app is reloaded(typically for app harness tests), we should maintain
             # the latest render function of a route.This applies typically to decorated pages
diff --git a/reflex/config.py b/reflex/config.py
index 7b910e3e..fcf29d5f 100644
--- a/reflex/config.py
+++ b/reflex/config.py
@@ -486,7 +486,7 @@ class EnvironmentVariables:
     REFLEX_FRONTEND_ONLY: EnvVar[bool] = env_var(False)
 
     # Reflex internal env to reload the config.
-    RELOAD_CONFIG: EnvVar[bool] = env_var(False, internal=True)
+    RELOAD_CONFIG: EnvVar[Optional[bool]] = env_var(None, internal=True)
 
     # If this env var is set to "yes", App.compile will be a no-op
     REFLEX_SKIP_COMPILE: EnvVar[bool] = env_var(False, internal=True)

```

Reply:
I will just assume that this was the correct behavior. I slightly improved the EnvVar api for such cases by adding `EnvVar.is_set`